server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -399,9 +399,8 @@ public MajorCompactor(ServerContext context) {
     public void run() {
       while (true) {
         try {
-          // TODO this property is misnamed, opened #3606
-          sleepUninterruptibly(getConfiguration().getTimeInMillis(Property.TSERV_MAJC_DELAY),
-              TimeUnit.MILLISECONDS);
+          // TODO Should there be a delay here? If so, what property should
```

Comment:
May be able to remove this entire while loop and put its code in a function that is called whenever closedLogs is modified.   Could have a function like the following and call it in the few places where closedLogs is modified.  One thing I am not sure about is if this code running inline after updating closedLogs instead of in another thread will cause problems, going to look into that a bit more.


```java
private void checkIfMinorCompactionsNeededForLogs(){
          final List<DfsLogger> closedCopy;

          synchronized (closedLogs) {
            closedCopy = List.copyOf(closedLogs);
          }
          // bail early now if we're shutting down
          for (Entry<KeyExtent,Tablet> entry : getOnlineTablets().entrySet()) {
            Tablet tablet = entry.getValue();
            tablet.checkIfMinorCompactionNeededForLogs(closedCopy);
          }
}
```

Reply:
I looked into moving the code and removing the thread in the tserver and made these changes https://github.com/keith-turner/accumulo/commit/3d8f6e232fc153ca9f869292536d9f49b500d204  

I think these changes are good.  That code that was executing in that thread only needs to run when something is added to the closed logs and additions only happen in one place, so moved the code there.